File: libnetwork/src/main/java/com/apache/fastandroid/network/retrofit/HeaderInterceptor.java
package com.apache.fastandroid.network.retrofit;

import android.os.Build;

import com.tesla.framework.common.util.log.NLog;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Created by Jerry on 2021/3/23.
 */
public class HeaderInterceptor implements Interceptor {
    public static final String TAG = HeaderInterceptor.class.getSimpleName();
    @NotNull
    @Override
    public Response intercept(@NotNull Chain chain) throws IOException {
        Request request = chain.request();
        String url = request.url().toString();


        Request.Builder builder = chain.request().newBuilder();
        try {
            //添加公共请求头
            builder.addHeader("brand", Build.BRAND)
                    .addHeader("model", Build.MODEL);
        } catch (Exception e) {
            NLog.printStackTrace(TAG, e);
        }
        return chain.proceed(builder.build());
    }
}

File: libnetwork/src/main/java/com/apache/fastandroid/network/interceptor/HeaderInterceptor.kt
package com.apache.fastandroid.network.interceptor

import android.os.Build

import okhttp3.*
import java.io.IOException



/**
 * Created by Jerry on 2021/3/23.
 */
class HeaderInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val originRequest: Request = chain.request()
        //添加公共请求头
        val newBuilder = originRequest.newBuilder()
            .apply {
                addHeader("brand", Build.BRAND)
                addHeader("model", Build.MODEL)
                addHeader("Content-Type", "application/json")
                addHeader("charset", "UTF-8")
            }

        return chain.proceed(newBuilder.build())
    }

    companion object {
        val TAG = HeaderInterceptor::class.java.simpleName
    }
}